### ISSUE TITLE
Can't query users by role with 'list_users' capability added

### ISSUE THREAD

**Issue by mruknowme** [wp-graphql/wp-graphql#2576]
### Description

The following query:
```graphql
query Staff {
  users(where: {role: SHOP_MANAGER}) {
    nodes {
      username
    }
  }
}
```
Works for Admin but fails for Shop Manager with error `Field \"users\" argument \"where\" requires type UserRoleEnum, found SHOP_MANAGER.` (Only allows `CUSTOMER` as enum value for Shop Manager)

`list_users` capability is added to Shop Manager
```php
<?

add_action( 'init', 'add_shop_manager_user_listing_capability');
function add_shop_manager_user_listing_capability() {
	$shop_manager = get_role( 'shop_manager' );
	$shop_manager->add_cap( 'list_users' );
}
```

### Steps to reproduce

Above

### WPGraphQL Version

1.12.0

### WordPress Version

6.0.2

### PHP Version

7.4.x

### Additional enviornment details

Woocommerce 6.9.4
WooGraphQL 0.11.1

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [ ] Yes
- [X] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by justlevine** [comment]
related: https://github.com/wp-graphql/wp-graphql/issues/2537#issuecomment-1256965313

**Comment by justlevine** [comment]
I believe this was fixed by #2967, since `UserRoleEnum` is now static and doesn't change by user.

Needs confirmation before closing

**Comment by justlevine** [comment]
Confirmed fixed in [v1.17.0](https://github.com/wp-graphql/wp-graphql/releases/tag/v1.17.0) 🚀
